cli/config/configfile/file.go

```diff
@@ -268,7 +273,15 @@ func (configFile *ConfigFile) GetAuthConfig(serverAddress string) (types.AuthCon
 // configured.
 func getConfiguredCredentialStore(c *ConfigFile, serverAddress string) string {
 	if c.CredentialHelpers != nil && serverAddress != "" {
-		if helper, exists := c.CredentialHelpers[serverAddress]; exists {
+		// Strip the scheme from the serverAddress, if present.
+		stripped := serverAddress
+		if strings.HasPrefix(serverAddress, "http://") {
+			stripped = strings.TrimPrefix(serverAddress, "http://")
+		} else if strings.HasPrefix(serverAddress, "https://") {
+			stripped = strings.TrimPrefix(serverAddress, "https://")
+		}
```

Comment:
This should probably use `registry.ConvertToHostname()`, although I'm not sure this is the correct place to do the convert.

How is this related to the rest of the change?